### PR TITLE
initial impl

### PR DESCRIPTION
**Pull request by konrad0960**

### CODE REVIEW

File: cancer_ai/validator/model_runners/onnx_runner.py [hunk 1]
```diff
@@ -52,3 +81,30 @@ async def run(self, preprocessed_data_generator: AsyncGenerator[np.ndarray, None
...
+                gender = 1
+            elif gender_str in ['female', 'f']:
+                gender = 0
+            else:
+                gender = -1  # Unknown/missing gender
+            
+            # Convert location to numerical: arm=1, feet=2, genitalia=3, hand=4, head=5, leg=6, torso=7, unknown=-1
+            location_str = entry.get('location', '').lower() if entry.get('location') else ''
+            location_mapping = {
+                'arm': 1, 'feet': 2, 'genitalia': 3, 'hand': 4,
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/model_runners/onnx_runner.py]: use enums

File: cancer_ai/validator/competition_manager.py [hunk 2]
```diff
@@ -163,11 +163,17 @@ async def evaluate(self) -> Tuple[str | None, BaseModelEvaluationResult | None]:
 
         
         await self.dataset_manager.prepare_dataset()
-        X_test, y_test = await self.dataset_manager.get_data()
+        X_test, y_test, metadata = await self.dataset_manager.get_data()
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/competition_manager.py:166]: metadata should be included in X_test, not separate variable (it was like this with melanoma)

File: cancer_ai/validator/dataset_manager.py [hunk 3]
```diff
@@ -137,10 +158,15 @@ async def prepare_dataset(self) -> None:
...
         """Get data from dataset handler"""
         if not self.data:
             raise DatasetManagerException(
                 f"Dataset '{self.competition_id}' not initalized "
             )
+        # Handle backward compatibility - if data has 2 elements, add empty metadata
+        if len(self.data) == 2:
+            x_data, y_data = self.data
+            metadata = [{'age': None, 'gender': None} for _ in x_data]
+            return x_data, y_data, metadata
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/dataset_manager.py:205]: metadata is together with X_data

File: cancer_ai/validator/model_runners/pytorch_runner.py [hunk 4]
```diff
@@ -1,16 +1,17 @@
 from . import BaseRunnerHandler
-from typing import List, AsyncGenerator
+from typing import List, AsyncGenerator, Union, Dict, Any, Tuple
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/model_runners/pytorch_runner.py]: revert whole file

File: cancer_ai/validator/model_runners/tensorflow_runner.py [hunk 5]
```diff
@@ -1,15 +1,16 @@
 from . import BaseRunnerHandler
-from typing import List, AsyncGenerator
+from typing import List, AsyncGenerator, Union, Dict, Any, Tuple
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/model_runners/tensorflow_runner.py]: revert whole file

File: cancer_ai/validator/dataset_handlers/image_csv.py [hunk 6]
```diff
@@ -30,31 +33,69 @@ def __init__(self, config, dataset_path, label_path: str) -> None:
         self.config = config
         self.dataset_path = dataset_path
         self.label_path = label_path
-        self.metadata_columns = ["filepath", "is_melanoma"]
+        self.metadata_columns = ["filepath", "label", "age", "gender", "location"]
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/dataset_handlers/image_csv.py]: TODO later leave it generic, it should return 

```
class ImageEntry:
    relative_path: str
    label: List[Any] 
```

and label should be processed in in `preprocess_and_serialize_data`

File: cancer_ai/validator/model_runners/pytorch_runner.py (hunk 4 above)
Comment (konrad0960) [reply cancer_ai/validator/model_runners/pytorch_runner.py]: why tho, i know we dont use it but why not to be aligned with other runners

File: cancer_ai/validator/model_runners/tensorflow_runner.py (hunk 5 above)
Comment (konrad0960) [reply cancer_ai/validator/model_runners/tensorflow_runner.py]: same as above - u mean to completely delte tensorflow and pytorch runners or what?

File: cancer_ai/validator/dataset_manager.py [hunk 7]
```diff
@@ -113,15 +113,36 @@ def set_dataset_handler(self) -> None:
             raise DatasetManagerException(
                 f"Dataset '{self.config.competition_id}' not downloaded"
             )
-        # is csv in directory
-        if os.path.exists(Path(self.local_extracted_dir, "labels.csv")):
+        
+        # Look for labels.csv in the extracted directory or its subdirectories
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/dataset_manager.py]: why it should look in subdirectories? labels.csv should exist in root dir, no need to look further

Comment (konrad0960) [reply cancer_ai/validator/dataset_manager.py]: i had some issues with current impl and this fixes that

Comment (konrad0960) [reply cancer_ai/validator/dataset_manager.py]: u can try to run it on ur side without it and see if it works, not in my case

File: cancer_ai/validator/model_runners/pytorch_runner.py (hunk 4 above)
Comment (wjurkowlaniec) [reply cancer_ai/validator/model_runners/pytorch_runner.py]: we don't have other runners, please revert. also this logic shouldn't be there, it should be in competition-speficif handler. this is the problem I solved with my original PR and let's no go back

File: cancer_ai/validator/model_runners/tensorflow_runner.py (hunk 5 above)
Comment (wjurkowlaniec) [reply cancer_ai/validator/model_runners/tensorflow_runner.py]: revert to previous stage, from my original PR

File: cancer_ai/validator/competition_manager.py (hunk 2 above)
Comment (konrad0960) [reply cancer_ai/validator/competition_manager.py:166]: Would be less clear separation of concerns, Id consider current approach and upgrade

File: cancer_ai/validator/dataset_manager.py (hunk 3 above)
Comment (konrad0960) [reply cancer_ai/validator/dataset_manager.py:205]: its better to have separation of concerns
more clear

File: cancer_ai/validator/model_runners/onnx_runner.py (hunk 1 above)
Comment (konrad0960) [reply cancer_ai/validator/model_runners/onnx_runner.py]: ok

File: cancer_ai/validator/dataset_handlers/image_csv.py (hunk 6 above)
Comment (konrad0960) [reply cancer_ai/validator/dataset_handlers/image_csv.py]: What would be in the List? Multiple labels per image?
loses type safety, not good

File: cancer_ai/validator/model_runners/onnx_runner.py [hunk 8]
```diff
@@ -52,3 +81,30 @@ async def run(self, preprocessed_data_generator: AsyncGenerator[np.ndarray, None
...
+    def _prepare_metadata_array(self, metadata: List[Dict[str, Any]]):
+        """Convert metadata list to numpy array for ONNX model input"""
+        # Convert metadata to numerical format
+        metadata_array = []
+        for entry in metadata:
+            age = entry.get('age', 0) if entry.get('age') is not None else 0
+            # Convert gender to numerical: male=1, female=0, unknown=-1
+            gender_str = entry.get('gender', '').lower() if entry.get('gender') else ''
+            if gender_str in ['male', 'm']:
+                gender = 1
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/model_runners/onnx_runner.py:94]: here as well

Comment (konrad0960) [reply cancer_ai/validator/model_runners/onnx_runner.py:94]: ale co

File: cancer_ai/validator/model_runners/pytorch_runner.py (hunk 4 above)
Comment (konrad0960) [reply cancer_ai/validator/model_runners/pytorch_runner.py]: ok

File: cancer_ai/validator/model_runners/tensorflow_runner.py (hunk 5 above)
Comment (konrad0960) [reply cancer_ai/validator/model_runners/tensorflow_runner.py]: ok

File: cancer_ai/validator/competition_handlers/tricorder_handler.py [hunk 9]
```diff
@@ -392,15 +488,19 @@ def _calculate_risk_category_scores(self, f1_scores: np.ndarray) -> Dict[RiskCat
 
     def _calculate_weighted_f1(self, category_scores: Dict[RiskCategory, float]) -> float:
         """Calculate weighted F1 score based on risk categories."""
-        total_weight = sum(info["weight"] for info in CLASS_INFO.values())
+        # Use category-level weights: HIGH_RISK=3.0, MEDIUM_RISK=2.0, BENIGN=1.0
+        category_weights = {
```
Comment (wjurkowlaniec) [inline cancer_ai/validator/competition_handlers/tricorder_handler.py:516]: here

Comment (wjurkowlaniec) [reply cancer_ai/validator/competition_handlers/tricorder_handler.py:516]: please don't put numbers in logic

Comment (wjurkowlaniec) [reply cancer_ai/validator/competition_handlers/tricorder_handler.py:516]: on top
